homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,217 @@
+"""
+Support for Honeywell Lyric devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/lyric/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import CONF_SCAN_INTERVAL, HTTP_BAD_REQUEST
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ["http"]
+
+DOMAIN = "lyric"
+
+DATA_LYRIC = "lyric"
+
+LYRIC_CONFIG_FILE = "lyric.conf"
+CONF_CLIENT_ID = "client_id"
+CONF_CLIENT_SECRET = "client_secret"
+CONF_REDIRECT_URI = "redirect_uri"
+CONF_LOCATIONS = "locations"
+CONF_FAN = "fan"
+CONF_AWAY_PERIODS = "away_periods"
+
+DEFAULT_FAN = False
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_CLIENT_ID): cv.string,
+                vol.Required(CONF_CLIENT_SECRET): cv.string,
+                vol.Optional(CONF_REDIRECT_URI): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=270): cv.positive_int,
+                vol.Optional(CONF_LOCATIONS): vol.All(cv.ensure_list, cv.string),
+                vol.Optional(CONF_FAN, default=DEFAULT_FAN): cv.boolean,
+                vol.Optional(CONF_AWAY_PERIODS): vol.All(cv.ensure_list, cv.string),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def request_configuration(lyric, hass, config):
+    """Request configuration steps from the user."""
+
+    configurator = hass.components.configurator
+    if "lyric" in _CONFIGURING:
+        _LOGGER.debug("configurator failed")
+        configurator.notify_errors(
+            _CONFIGURING["lyric"], "Failed to configure, please try again."
+        )
+        return
+
+    def lyric_configuration_callback(data):
+        """Run when the configuration callback is called."""
+
+        _LOGGER.debug("configurator callback")
+        # url = data.get('url')
+        setup_lyric(hass, lyric, config)  # url=url
+
+    hass.http.register_view(LyricAuthenticateView(lyric))
+
+    _CONFIGURING["lyric"] = configurator.request_config(
```

Comment:
The configurator is deprecated. Use a config flow instead.

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler#defining-your-config-flow>

and for handling OAuth:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler#configuration-via-oauth2>